File: Beginner/Grade The Steel/gradeTheSteel.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";
process.stdin.on('data', function(chuck) {
    input += chuck.toString();
});

process.stdin.on('end', function() {
    gradeTheSteel(input);
});

function gradeTheSteel(args) {
    let arr = args.split("\n");
    let n = arr[0];
    for (let i=1; i<=n; i++) {
        [hardness, carbon, tensile] = arr[i].split(" ").map((x) => +x);
        if (hardness > 50 && carbon < 0.7 && tensile > 5600) {
	        console.log(10);
	    } else if (hardness > 50 && carbon < 0.7) {
	        console.log(9);
	    } else if (carbon < 0.7 && tensile > 5600) {
	        console.log(8);
	    } else if (hardness > 50 && tensile > 5600) {
	        console.log(7);
	    } else if (hardness > 50 || carbon < 0.7 || tensile > 5600) {
	        console.log(6);
	    } else {
	        console.log(5);
	    }
    }
}

File: Beginner/Grade The Steel/GradeTheSteel.py
n = int(input())
l = []
for i in range(n):
    hardness, carbon, tensile = map(float, input().split())
    if hardness > 50 and carbon < 0.7 and tensile > 5600:
        result = 10
    elif hardness > 50 and carbon < 0.7:
        result = 9
    elif carbon < 0.7 and tensile > 5600:
        result = 8
    elif hardness > 50 and tensile > 5600:
        result = 7
    elif hardness > 50 or carbon < 0.7 or tensile > 5600:
        result = 6
    else:
        result = 5
    print(result)
